zio/src/test/scala/ReadmeRioConstructionCodeSnippetsTest.scala

```diff
@@ -43,60 +43,75 @@ import scala.annotation.nowarn
       import Setup._
 
       // Case 1: from a possible config in a Layer (gives a Layer)
-      val log4sCase1: RLayer[Has[AConfig], ZLogWriter] =
-        ZLayer.fromServiceM { config =>
-          log4sFromName.provide(config.logName)
-        }
-      val scribe4sCase1: RLayer[Has[AConfig], ZLogWriter] =
-        ZLayer.fromServiceM { config =>
-          scribeFromName.provide(config.logName)
-        }
+      val log4sCase1: RLayer[AConfig, ZLogWriter] = ZLayer {
+        ZIO.service[AConfig].flatMap(c => log4sFromName.provideEnvironment(ZEnvironment(c.logName)))
```

Comment:
serviceWithZio could work here
```suggestion
        ZIO.serviceWithZIO(c => log4sFromName.provideEnvironment(ZEnvironment(c.logName)))
```